tests/infra/proposal_generator.py

```diff
@@ -0,0 +1,336 @@
+        return build_proposal("trust_node", node_id, **kwargs)
+
+    @staticmethod
+    def retire_node_proposal(node_id, **kwargs):
+        return build_proposal("retire_node", node_id, **kwargs)
+
+    @staticmethod
+    def new_node_code_proposal(code_digest, **kwargs):
+        if isinstance(code_digest):
+            code_digest = list(bytearray.fromhex(code_digest))
+        return build_proposal("new_node_code", code_digest, **kwargs)
+
+    @staticmethod
+    def new_user_code_proposal(code_digest, **kwargs):
+        if isinstance(code_digest):
+            code_digest = list(bytearray.fromhex(code_digest))
+        return build_proposal("new_user_code", code_digest, **kwargs)
+
+    @staticmethod
+    def accept_recovery_proposal(**kwargs):
+        return build_proposal("accept_recovery", **kwargs)
+
+    @staticmethod
+    def open_network_proposal(**kwargs):
+        return build_proposal("open_network", **kwargs)
+
+    @staticmethod
+    def rekey_ledger_proposal(**kwargs):
+        return build_proposal("rekey_ledger", **kwargs)
+
+    @staticmethod
+    def update_recovery_shares_proposal(**kwargs):
+        return build_proposal("update_recovery_shares", **kwargs)
+
+    @staticmethod
+    def set_recovery_threshold_proposal(threshold, **kwargs):
+        return build_proposal("set_recovery_threshold", threshold, **kwargs)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+
+    default_proposal_output = "{proposal_type}.json"
+    default_vote_output = "vote_for_{proposal_output}.json"
+
+    parser.add_argument(
+        "-po",
+        "--proposal-output-file",
+        type=str,
+        help=f"Path where proposal JSON object (request body for /gov/propose)"
+        "will be dumped. Default is {default_proposal_output}",
+    )
+    parser.add_argument(
+        "-vo",
+        "--vote-output-file",
+        type=str,
+        help=f"Path where vote JSON object (request body for /gov/vote) will be"
+        "dumped. Default is {default_vote_output}",
+    )
+    parser.add_argument(
+        "-pp",
+        "--pretty-print",
+        action="store_true",
+        help="Pretty-print the JSON output",
+    )
+    parser.add_argument(
+        "-i",
+        "--inline-args",
+        action="store_true",
+        help="Create a fixed proposal script with the call arguments as literalsinside"
+        "the script. When not inlined, the parameters are passed separately and could"
+        "be replaced in the resulting object",
+    )
+
+    # Auto-generate CLI args based on the inspected generator signatures
+    proposal_generators = inspect.getmembers(Proposals, predicate=inspect.isfunction)
+    subparsers = parser.add_subparsers(
+        title="Possible proposals", dest="proposal_type", required=True
+    )
+
+    for func_name, func in proposal_generators:
+        suffix = "_proposal"
```

Comment:
We can avoid that kludge if we explicitly decorate them (and shorten the names!).